plugins/plugin-testing-java/plugin-testing-testng/che-plugin-testing-testng-server/src/main/java/org/eclipse/che/plugin/testing/testng/server/TestNGRunner.java

```diff
@@ -40,8 +71,114 @@ public TestResult execute(Map<String, String> testParameters) throws Exception {
     }
 
     @Override
+    @Nullable
     public ProcessHandler execute(TestExecutionContext context) {
+        String projectPath = context.getProjectPath();
+        IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectPath);
+        IJavaProject javaProject = JavaModelManager.getJavaModelManager().getJavaModel().getJavaProject(project);
+        if (javaProject.exists()) {
+            return startTestProcess(javaProject, context);
+        }
+
+        return null;
+    }
+
+    private ProcessHandler startTestProcess(IJavaProject javaProject, TestExecutionContext context) {
+        File suiteFile = createSuite(context, javaProject);
+        if (suiteFile == null) {
+            throw new RuntimeException("Can't create TestNG suite xml file.");
+        }
+
+        JavaParameters parameters = new JavaParameters();
+        parameters.setJavaExecutable("java");
+        parameters.setMainClassName("org.testng.CheTestNGLauncher");
+        parameters.setWorkingDirectory("/tmp");
```

Comment:
Several testing frameworks or test runners (such as arquillian, etc ...) assume that the current working directory should be the root directory of the project.
See [this issue](https://issues.jboss.org/browse/CHE-214?focusedCommentId=13397893&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13397893) as an example.
So it seems that at this line the working directory should be set to the root directory of the project being tested, instead of `/tmp`.
This would also be consistent with the way tests are run inside maven afaik.